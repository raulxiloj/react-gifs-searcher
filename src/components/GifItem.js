import React from 'react'
import PropTypes from 'prop-types'

export const GifItem = ({ title, url }) => {

    return (
        <div className="col-md-3 mb-4 align-items-stretch animate__animated animate__fadeIn">
            <div className="card" >

                <img className="card-img-top" src={url} alt={title} />

                <div className="card-body">
                    <p className="card-text">{title}</p>
                </div>

            </div>
        </div>
    )
}

GifItem.propTypes = {
    title: PropTypes.string.isRequired,
    url: PropTypes.string.isRequired
}