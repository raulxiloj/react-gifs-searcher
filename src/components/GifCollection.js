import React from 'react'
import PropTypes from 'prop-types';
import { GifItem } from './GifItem';

import { useFetchGifs } from '../hooks/useFetchGifs';

export const GifCollection = ({ category }) => {//Gif grid

    const { data: images, loading } = useFetchGifs(category);

    return (
        <>
            <h3 className="animate__animated animate__fadeIn">{category}</h3>
            {loading && <p className="animate__animated animate__flash">Loading...</p>}
            <div className="card-grid animate__animated animate__fadeIn">
                {
                    images.map(img => <GifItem key={img.id} {...img} />)
                }
            </div>
        </>
    )
}

GifCollection.propTypes = {
    category: PropTypes.string.isRequired
}