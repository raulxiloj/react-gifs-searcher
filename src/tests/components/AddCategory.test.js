import React from 'react';
import '@testing-library/jest-dom'
import { shallow } from "enzyme"
import { AddCategory } from "../../components/AddCategory"

describe('Tests in AddCategory component', () => {

    const setCategories = jest.fn();
    let wrapper = shallow(<AddCategory setCategories={setCategories} />);

    beforeEach(() => {
        jest.clearAllMocks();
        wrapper = shallow(<AddCategory setCategories={setCategories} />);
    })

    test('It should show the component correctly', () => {
        expect(wrapper).toMatchSnapshot();
    })

    test('It should change the input text', () => {
        const input = wrapper.find('input');
        const value = 'Hello world';
        input.simulate('change', { target: { value } });
    })

    test('It shouldnt post the data with submit', () => {
        wrapper.find('form').simulate('submit', { preventDefault: () => { } });
        expect(setCategories).not.toHaveBeenCalled();
    })

    test('It should call the function setCategories and clean the input', () => {
        const input = wrapper.find('input');
        const value = 'Relajate medusin';
        input.simulate('change', { target: { value } });
        wrapper.find('form').simulate('submit', { preventDefault: () => { } });
        expect(setCategories).toHaveBeenCalled();
        //expect(setCategories).toHaveBeenCalledTimes(1);
        expect(input.props().value).toBe('');
    })



})
