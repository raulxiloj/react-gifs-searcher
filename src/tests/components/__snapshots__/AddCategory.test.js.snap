// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Tests in AddCategory component It should show the component correctly 1`] = `
<form
  onSubmit={[Function]}
>
  <input
    onChange={[Function]}
    placeholder="Search something"
    type="text"
    value=""
  />
</form>
`;
